primefaces/src/main/java/org/primefaces/component/api/DialogReturnAware.java

```diff
@@ -44,7 +44,7 @@ default boolean isDialogReturnEvent(FacesEvent event, FacesContext context) {
             Map<String, String> params = context.getExternalContext().getRequestParameterMap();
             String eventName = params.get(Constants.RequestParams.PARTIAL_BEHAVIOR_EVENT_PARAM);
 
-            return eventName.equals(EVENT_DIALOG_RETURN);
+            return EVENT_DIALOG_RETURN.equals(eventName);
```

Comment:
This work actually uncovered an unrelated bug here. This line threw NPE when `DynamicOncompleteEvent` was queued but it could as good happen for any other custom AjaxBehaviorEvent.